src/OpenTelemetry.Contrib.Instrumentation.AWS/Implementation/AWSTracingPipelineHandler.cs

```diff
@@ -0,0 +1,270 @@
+        private static readonly Action<IDictionary<string, string>, string, string> Setter = (carrier, name, value) =>
+        {
+            carrier[name] = value;
+        };
+
+        private static readonly Dictionary<string, string> ServiceParameterMap = new Dictionary<string, string>()
+        {
+            { "DynamoDBv2", "TableName" },
+            { "SQS", "QueueUrl" },
+        };
+
+        private static readonly Dictionary<string, string> ParameterAttributeMap = new Dictionary<string, string>()
+        {
+            { "TableName", AWSSemanticConventions.AttributeAWSDynamoTableName },
+            { "QueueUrl", AWSSemanticConventions.AttributeAWSSQSQueueUrl },
+        };
+
+        private static readonly ActivitySource AWSSDKActivitySource = new ActivitySource(ActivitySourceName);
+
+        private readonly AWSClientInstrumentationOptions options;
+
+        public AWSTracingPipelineHandler(AWSClientInstrumentationOptions options)
+        {
+            this.options = options;
+        }
+
+        public override void InvokeSync(IExecutionContext executionContext)
+        {
+            var activity = this.ProcessBeginRequest(executionContext);
+            try
+            {
+                base.InvokeSync(executionContext);
+            }
+            catch (Exception ex)
+            {
+                if (activity != null)
+                {
+                    this.ProcessException(activity, ex);
+                }
+
+                throw;
+            }
+            finally
+            {
+                if (activity != null)
+                {
+                    this.ProcessEndRequest(executionContext, activity);
+                }
+            }
+        }
+
+        public override async Task<T> InvokeAsync<T>(IExecutionContext executionContext)
+        {
+            T ret = null;
+
+            var activity = this.ProcessBeginRequest(executionContext);
+            try
+            {
+                ret = await base.InvokeAsync<T>(executionContext).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                if (activity != null)
+                {
+                    this.ProcessException(activity, ex);
+                }
+
+                throw;
+            }
+            finally
+            {
+                if (activity != null)
+                {
+                    this.ProcessEndRequest(executionContext, activity);
+                }
+            }
+
+            return ret;
+        }
+
+        private Activity ProcessBeginRequest(IExecutionContext executionContext)
+        {
+            Activity activity = null;
+
+            var requestContext = executionContext.RequestContext;
+            var service = this.GetAWSServiceName(requestContext);
+            var operation = this.GetAWSOperationName(requestContext);
+
+            activity = AWSSDKActivitySource.StartActivity(service + "." + operation, ActivityKind.Client);
```

Comment:
nit: you mayoptimize perf by setting activity.DisplayName = service + "." + operation, after starting the activity, and know activity is sampled in, assuming sampler's don't make decision based on display name.

Reply:
Would that be a really big perf improvement given that it would involve conditional check and name assignment afterward? If it's not a big difference, I would prefer to keep it this way for simplicity.

Reply:
And I believe that the `ActivitySource.StartActivity` always requires the activity name as an argument. Right?

Reply:
It depends on sampling rate. The current code always allocates the string. If you set the name *after* knowing sampling decision, you pay for the allocation only for activities which are going to be exported and not dropped on the floor.

(DisplayName can be changed anytime)

Reply:
I see. I can do this optimization in subsequent PRs.